MailCore/Cache/RefreshAppBackgroundTask.swift

```diff
@@ -64,18 +65,53 @@ public class RefreshAppBackgroundTask: BackgroundTaskable {
     }
 
     public func run() async {
-        guard let currentMailboxManager = accountManager.currentMailboxManager,
-              let inboxFolder = currentMailboxManager.getFolder(with: .inbox),
-              inboxFolder.cursor != nil else {
-            // We do nothing if we don't have an initial cursor
+        guard !accountManager.accounts.isEmpty else {
```

Comment:
Debatable but I prefer this. What do you think ?
```suggestion
        guard accountManager.currentAccount != nil else {
```

Reply:
Okay, so if `AccountManageable` encapsulates the management of user accounts, it should expose a `someUserAuthenticated` so less leaks to this component.
I feel I could make your own argument with checking the array of accounts directly also.
I'm ok with this, resolving now.